utils/patientmatching/rule_based_patient_matcher.bal

```diff
@@ -0,0 +1,282 @@
+            username: <string>masterPatientIndexDbUser,
+            password: <string>masterPatientIndexDbPassword
+        };
+        return mpiDbConfig;
+    }
+
+    isolated function getSQLQuery(r4:Patient patient, RulesRecord rulesTable, ConfigurationRecord config) returns (error|string) {
+        json|error MPIColumnNames = config?.masterPatientIndexColumnNames;
+        json|error MPITableName = config?.masterPatientIndexTableName;
+        if MPIColumnNames is error {
+            return throwFHIRError("Error occurred while getting the MPIColumnNames from the config.json file.", cause = MPIColumnNames);
+        }
+        if MPITableName is error {
+            return throwFHIRError("Error occurred while getting the MPITableName from the config.json file.", cause = MPITableName);
+        }
+        json[] parameters = <json[]>MPIColumnNames;
+        string tableName = <string>MPITableName;
+        string query = "SELECT * FROM ";
+        query = string `${query} ${tableName} WHERE `;
+        int lastIndex = parameters.length() - 1;
+        foreach int i in 0 ... parameters.length() {
+            string fhirPathRule = rulesTable.fhirpathArray[i];
+            fhirpath:FhirPathResult fhirPathResult = fhirpath:getFhirPathResult(<map<json>>patient.toJson(), fhirPathRule);
+            if i == lastIndex {
+                string fhirPathRuleLastIndex = rulesTable.fhirpathArray[i];
+                map<anydata> resultMapLastIndex = fhirpath:getFhirPathResult(<map<json>>patient.toJson(), fhirPathRuleLastIndex);
+                if resultMapLastIndex.hasKey(RESULT) {
+                    json resultValue = <json>fhirPathResult.get(RESULT);
+                    if resultValue is json[] {
+                        json resultFirstElement = resultValue[0];
+                        string queryParam = string `${"\""}${resultFirstElement.toString()}${"\""}`;
+                        query = string ` ${query} ${parameters[i].toString()} = ${queryParam}`;
+                    } else {
+                        string queryParam = string `${"\""}${resultValue.toString()}${"\""}`;
+                        query = string `${query} ${parameters[i].toString()} = ${queryParam}`;
+                    }
+                    break;
+                } else {
+                    return fhirpath:createFhirPathError("No result found for the given FHIRPath expression in the patient: ", fhirPathRuleLastIndex);
+                }
+            }
+            if fhirPathResult.hasKey(RESULT) {
+                json resultPath = <json>fhirPathResult.get(RESULT);
+                if fhirPathResult.get(RESULT) is json[] {
+                    json[] resultValue = <json[]>fhirPathResult.get(RESULT);
+                    json resultFirstElement = resultValue[0];
+                    string queryParam = string `${"\""}${resultFirstElement.toString()}${"\""}`;
+                    query = string `${query} ${parameters[i].toString()} = ${queryParam} AND `;
```

Comment:
We can extract out this line since it is common for both if and else block

Reply:
Fixed in bd514ee7ad5e188fbaa6a4484af40fc238568f8d